pkg/virt-handler/vm.go

```diff
@@ -922,10 +922,11 @@ func (d *VirtualMachineController) updateAccessCredentialConditions(vmi *v1.Virt
 }
 
 func (d *VirtualMachineController) updateLiveMigrationConditions(vmi *v1.VirtualMachineInstance, condManager *controller.VirtualMachineInstanceConditionManager) {
-
-	// Cacluate whether the VM is migratable
-	liveMigrationCondition, isBlockMigration := d.calculateLiveMigrationCondition(vmi)
-	if !condManager.HasCondition(vmi, v1.VirtualMachineInstanceIsMigratable) {
+	liveMigrationCondition := condManager.GetCondition(vmi, v1.VirtualMachineInstanceIsMigratable)
+	isConditionSet := liveMigrationCondition != nil
+	// Calcuate whether the VM is migratable
+	liveMigrationCondition, isBlockMigration := d.calculateLiveMigrationCondition(vmi, liveMigrationCondition)
+	if !isConditionSet {
```

Comment:
Instead of adding another nil condition check to `calculateLiveMigrationCondition()`, why not just move the call to `calculateLiveMigrationCondition()` under the `if !condManager.HasCondition()` block?
Something like:
```
// Calcuate whether the VM is migratable
if !condManager.HasCondition(vmi, v1.VirtualMachineInstanceIsMigratable) {
	liveMigrationCondition, isBlockMigration := d.calculateLiveMigrationCondition(vmi)
```

Reply:
> Ah, got it, thanks! One last nit, `calculateLiveMigrationCondition()` seems to return exactly the condition that was passed to it. Maybe it doesn't need to return the condition anymore?

Nevermind, I missed the part where the condition might actually get modified, sorry about that!
/approve